modules/battery_manager/php/testcontroller.class.inc

```diff
@@ -0,0 +1,205 @@
+    ) : ResponseInterface {
+        return $handler->handle($request);
+    }
+
+    /**
+     * This acts as an Ajax enpoint that handles all action requests from the
+     * Battery Manager Module.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    protected function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $this->db   = \Database::singleton();
+        $this->user = $request->getAttribute('user');
+        $method     = $request->getMethod();
+        $testArray  = json_decode($request->getBody()->getContents(), true);
+        $test       = new Test($testArray);
+
+        switch($method) {
+        case 'GET':
+            return $this->getInstances();
+        case 'POST':
+            return $this->postInstance($test);
+        case 'PUT':
+            return $this->putInstance($test);
+        }
+    }
+
+    /**
+     * Gets the data source for battery manager tests.
+     *
+     * @return \LORIS\Data\Provisioner
+     */
+    private function getDataProvisioner() : \LORIS\Data\Provisioner
+    {
+        $provisioner = new TestProvisioner();
+
+        if ($this->user->hasPermission('access_all_profiles') == false) {
+            $provisioner = $provisioner->filter(
+                new \LORIS\Data\Filters\UserSiteMatch()
+            );
+        }
+        return $provisioner;
+    }
+
+    /**
+     * Updates the status of the given test from Active to Deactive or
+     * viceversa.
+     *
+     * @param \Test     $test Test to be saved
+     *
+     * @return \LORIS\Http\Response\JSON response
+     */
+    private function putInstance(\Test $test)
+    {
+        $testArray = $test->toSQL();
+        try {
+            $this->db->update('test_battery', $testArray, array('ID' => $testArray['ID']));
+            return new OK('Availability Updated Successfully');
+        } catch (\DatabaseException $e) {
+            return new InternalServerError(
+                'Could not update entry in the Test Battery'
+            );
+        }
+    }
+
+    /**
+     * Adds a new test to the test_battery in the database.
+     *
+     * @param \Test     $test Test to be posted
+     *
+     * @return \LORIS\Http\Response response
+     */
+    private function postInstance(\Test $test)
+    {
+        if (!$this->user->hasPermission('battery_manager_edit')) {
+            return new Forbidden('Edit Permission Denied');
+        }
+
+        if ($this->isDuplicate($test)) {
+            return new Conflict('This Test already exists in the database');
+        }
+
+        $test->row['active'] = 'Y';
+        $testArray = $test->toSQL();
+
+        try {
+            $this->db->insert('test_battery', $testArray);
+            return new OK('Test Submission Successful');
+        } catch (\DatabaseException $e) {
+            return new InternalServerError(
```

Comment:
Does this class exist? I thought you added it in another PR that isn't merged yet.

Reply:
I'm pretty what you're referencing are custom exceptions that I was trying to merge, not JSON responses.

All our JSON Response classes can be found here: 
https://github.com/aces/Loris/tree/master/src/Http/Response/JSON

Reply:
I think the JSON responses need to be fully qualified like `\LORIS\Http\Response\JSON\InternalServerError(...);`

Reply:
Right... yeah. I forgot about that.